app/routes.py

```diff
@@ -1,2 +1,258 @@
+from app.models.rental import Rental
+from app import db
+from datetime import datetime, timedelta
+import requests
+import os
 
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+
+
+@customers_bp.route("", methods= ["GET", "POST"])
+def handle_get_customers():
+    """
+    Returns data for a GET query and creates a new 
+    customer through a POST query
+    Planning to separate out into handle_get_customers 
+    and handle_post_customers
+    """
+    if request.method == "GET":
+        customers = Customer.query.all()
+        get_response =[]
+        for customer in customers:
+            get_response.append(customer.make_json())
+        return jsonify(get_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        
+        if "name" not in request_body.keys() or "postal_code" not in \
+            request_body.keys() or "phone" not in request_body.keys():
+            return make_response({"details": "Invalid data"}, 400)
+        
+        new_customer = Customer(name=request_body["name"],\
+            postal_code=request_body["postal_code"], phone_number=request_body["phone"],\
+                register_at=datetime.now())
+        
+        db.session.add(new_customer)
+        db.session.commit()
+        return make_response(new_customer.make_json(), 201)
+
+@customers_bp.route("/<id>", methods = ["GET", "PUT", "DELETE"])
+def handle_customer(id):
+    """
+    Takes in cutomer id  
+    Returns information about that customer (if it exists)
+    for a GET query.
+    Alters/ updates one customer's data and returns the updated
+    information. 
+    Deletes all  customer information at the given id number
+    Planning to separate out into individual functions by method 
+    """
+    customer = Customer.query.get(id)
+
+    if customer is None:
+        return make_response("", 404)
+
+    elif request.method == "GET":
+        return customer.make_json()
+    
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        
+        if "name" not in request_body.keys() \
+            or "postal_code" not in request_body.keys() \
+                or "phone" not in request_body.keys():
+                return make_response({"details": "Invalid data"}, 400) 
+        
+        customer.name = request_body["name"]
+        customer.postal_code = request_body["postal_code"]
+        customer.phone_number = request_body["phone"]
+        return customer.make_json()
+    
+    elif request.method == "DELETE":
+        db.session.delete(customer)
+        db.session.commit()
+        return customer.return_id()
+
+@customers_bp.route("/<id>/rentals", methods = ["GET"])
+def handle_customer_rental(id):
+    """
+    takes in customer id
+    returns a list of videos currently checked out by that customer
+    """
+    customer = Customer.query.get(id)
+    
+    if customer is None:
+        return make_response("", 404)
```

Comment:
We could use `get_or_404()` method here 
https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.BaseQuery.get_or_404